source/SIL.AppBuilder.Portal.Frontend/src/ui/components/inputs/multi-select/index.tsx

```diff
@@ -0,0 +1,79 @@
+import * as React from 'react';
+import { ResourceObject } from 'jsonapi-typescript';
+
+import { attributesFor, relationshipFor } from '@data';
+import { i18nProps } from '@lib/i18n';
+import { Checkbox } from 'semantic-ui-react';
+import ProductIcon from '@ui/components/product-icon';
+
+import { isEmpty } from '@lib/collection';
+
+interface IOwnProps<T> {
+  className?: string;
+  name: string;
+  list: T[];
+  selected: T[];
+  onChange: (el: T) => void;
+}
+
+export type IProps<T extends ResourceObject> =
+  & IOwnProps<T>
+  & i18nProps;
+
+export class MultiSelect extends React.Component<IProps<ResourceObject>> {
+
+  onChange = (element) => (e) => {
+    e.preventDefault();
+    this.props.onChange(element);
+  }
+
+  inSelectedList = (element) => {
+
+    const { selected, name } = this.props;
+
+    const el = selected.find(selectedItem => {
+      const { data } = relationshipFor(selectedItem, name);
+      return data.id === element.id;
+    });
+
+    return el !== undefined;
+  }
+
+  render() {
+
+    const { list, t } = this.props;
```

Comment:
Since the HOC returns different names ("productDefinitions", "stores", "products") how can make it generic

Reply:
pass `list={productDefinition}`?

I don't understand the issue. :-\

if the MultiSelect is for some reason in a HoC mountain, you could use recompose`s withProps: https://github.com/acdlite/recompose/blob/master/docs/API.md#withprops